### PR TITLE
Akka HTTP-based client with 'persistent' connections

### PR DESCRIPTION
**Pull request by raboof**
The first 2 tests succeed, the tests that require failover to
another server (as expected) still fail.

Requires https://github.com/akka/akka-http/pull/3685

### CODE REVIEW

File: interop-tests/src/test/scala/akka/grpc/scaladsl/NonBalancingIntegrationSpec.scala [hunk 1]
```diff
@@ -51,6 +60,27 @@ class NonBalancingIntegrationSpec extends AnyWordSpec with Matchers with BeforeA
...
+      for (i <- 1 to 50) {
+        client.sayHello(HelloRequest(s"Hello $i")).futureValue
+      }
+      server1.terminate(5.seconds).futureValue
+      // And restart
+      Http().newServerAt("127.0.0.1", server1.localAddress.getPort).bind(GreeterServiceHandler(service1)).futureValue
+
+      for (i <- 1 to 50) {
+        client.sayHello(HelloRequest(s"Hello $i")).futureValue
+      }
```
[the remark this answers is not in the document]
Comment (raboof) [reply interop-tests/src/test/scala/akka/grpc/scaladsl/NonBalancingIntegrationSpec.scala:79]: I wanted to simulate 'an actively used server', throwing more requests at it perhaps has a better chance of hitting corner cases ;)

**Review by ignasi35 — APPROVED** [review]
looking good

File: runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala [hunk 2]
```diff
@@ -53,33 +53,29 @@ object AkkaHttpClientUtils {
...
-      case Some(authority) =>
-        ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, port) => {
+    val random = new Random()
+    val clientConnectionSettings =
+      ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, _) => {
+        settings.overrideAuthority.foreach { authority =>
           assert(host == authority)
-          assert(port == target.port.getOrElse(settings.defaultPort))
+        }
+        settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds).map { resolved =>
```
Comment (jrudolph) [inline runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala:66]: Is that the part that fixes the fail-over?

Comment (raboof) [reply runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala:66]: Basically yes: by doing the discovery inside the resolver, it is performed again for each materialization of the connection. A bit coarse perhaps, but does the trick for now here.
